src/lib.rs

```diff
@@ -197,36 +215,95 @@ pub use crate::session::Client;
 
 /// A single element on the current page.
 #[derive(Clone, Debug, Serialize)]
-pub struct Element {
+pub struct Element<C>
+where
+    C: NewConnector + connect::Connect,
+{
     #[serde(skip_serializing)]
-    client: Client,
+    client: Client<C>,
     #[serde(flatten)]
     element: webdriver::common::WebElement,
+    #[serde(skip_serializing)]
+    _marker: marker::PhantomData<C>,
 }
 
 /// An HTML form on the current page.
 #[derive(Clone, Debug)]
-pub struct Form {
-    client: Client,
+pub struct Form<C>
+where
+    C: NewConnector + connect::Connect,
+{
+    client: Client<C>,
     form: webdriver::common::WebElement,
+    _marker: marker::PhantomData<C>,
 }
 
-impl Client {
-    /// Create a new `Client` associated with a new WebDriver session on the server at the given
+impl<C> Client<C>
+where
+    C: NewConnector + connect::Connect,
+{
+    /// Create a new [`Client`][crate::Client] associated with a new WebDriver session on the server at the given
     /// URL.
     ///
+    ///
+    ///
     /// Calls `with_capabilities` with an empty capabilities list.
     #[allow(clippy::new_ret_no_self)]
     pub async fn new(webdriver: &str) -> Result<Self, error::NewSessionError> {
```

Comment:
I think what we should do is have `new` use `#[cfg(feature = "hyper-tls")]`, and specifically construct a `Client<hyper_tls::HttpsConnector>`. That way there's still an easy way for users to create a client that "just works" if they don't care too much about the underlying TLS stack, and you can remove all the `<...>` annotations you've had to add to the examples.

Reply:
So just leave new only on feature `openssl-tls` or `rustls-tls` as a default and in the case that user does default-features = false then they have to use with_capabilities with type parameters?
Also we could just do type aliases for the two connectors, that way the user can just `use fantoccini::Rustls` or `use fantoccini::OpenSSL` and just use this as a type parameter without the need to add hyper and hyper_rustls or hyper_tls to their dependencies